core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -237,10 +271,7 @@ private[spark] class BlockManager(
   private var asyncReregisterTask: Future[Unit] = null
   private val asyncReregisterLock = new Object
 
-  // Field related to peer block managers that are necessary for block replication
-  @volatile private var cachedPeers: Seq[BlockManagerId] = _
-  private val peerFetchLock = new Object
-  private var lastPeerFetchTimeNs = 0L
+  private[storage] var peerProvider: BlockManagerPeerProvider = _
```

Comment:
I get that we need the blockManagerId which we don't have until initilization is completed so this needs to be a var, but I'm better not refactor this function out into its own class.